app/src/app/app.component.ts

```diff
@@ -16,24 +16,74 @@ import { Component } from '@angular/core';
 import { MatDialog } from '@angular/material/dialog';
 import { ModalComponent } from './components/specmath-modal/modal.component';
 import { OperationSet } from 'src/shared/interfaces';
+import { MediaMatcher } from '@angular/cdk/layout';
+import { Router, RouterEvent, NavigationEnd } from '@angular/router';
+import { filter } from 'rxjs/operators';
+
+type Route = 'home' | 'result' | 'defaults' | 'about';
+
+type Routes = {
+  [route in Route]: string;
+};
+
+const routes: Routes = {
+  home: '/',
+  result: '/result',
+  defaults: '/defaults',
+  about: '/about',
+};
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  styleUrls: ['./app.component.scss'],
 })
 export class AppComponent {
+  currentRoute = routes.home;
+  mobileQuery: MediaQueryList;
   operationSet: OperationSet = {
     specFiles: [],
     resultSpec: null,
-    valid: false
+    valid: false,
   };
 
-  constructor(readonly dialog: MatDialog) { }
+  constructor(
+    readonly router: Router,
+    readonly dialog: MatDialog,
+    media: MediaMatcher
+  ) {
+    this.mobileQuery = media.matchMedia('(max-width: 768px)');
```

Comment:
Nit: move the mobile breakpoint into a constant.

Could also make a service for this if it will be needed by other components in the future.

Reply:
Done 👍, I made a service for the SideNav which I'm using for interaction labels